api/src/main/java/org/bf2/operator/resources/v1alpha1/TlsKeyPair.java

```diff
@@ -19,24 +18,48 @@
 @JsonInclude(JsonInclude.Include.NON_NULL)
 public class TlsKeyPair {
 
-    @NotNull
+    //@NotNull
     private String cert;
-    @NotNull
+    //@NotNull
     private String key;
+    private SecretKeySelector certRef;
+    private SecretKeySelector keyRef;
 
     public String getCert() {
-        return cert;
+        if(cert != null){
+            return cert;
+        }
+        return certRef.getKey();
```

Comment:
I don't think this method should change, continue returning `cert` unconditionally.

Reply:
So the idea of returning `certRef.getKey()` is to not change with the secret retrieval process in Operator. Wherever we are trying `TlsKeyPair.getCert()` it would continue with the same. But do you think we should change the TlsKeyPair retrieval process everywhere? ( I mean now we do `getCertRef` instead `getCert `?)

Reply:
I don't think that would work. The operator code still needs to change to either copy the secrets to their current locations using the `SecretKeySelector`s or use the refs directly.